gov/governance/execute.gno

```diff
@@ -1,622 +1,311 @@
 	}
+	return hdlr.(ParameterHandler), nil
+}
 
-	now := uint64(time.Now().Unix())
-
-	config := GetConfigVersion(proposal.ConfigVersion)
-	votingEnd := proposal.ExecutionState.CreatedAt + config.VotingStartDelay + config.VotingPeriod
-	windowStart := votingEnd + config.ExecutionDelay
-	if now < windowStart {
-		panic(addDetailToError(
-			errBeforeProposalExecutionTime,
-			ufmt.Sprintf("execute.gno__Execute() || EXECUTION_WINDOW_NOT_STARTED (now(%d) < windowStart(%d))", now, windowStart),
-		))
+func (r *ParameterRegistry) Get(pkgPath, function string) (ParameterHandler, error) {
+	key := makeHandlerKey(pkgPath, function)
+	hdlr, exists := r.handlers.Get(key)
+	if !exists {
+		return nil, ufmt.Errorf("handler not found for %s", key)
 	}
+	return hdlr.(ParameterHandler), nil
+}
 
-	windowEnd := windowStart + config.ExecutionWindow
-	if now >= windowEnd {
-		panic(addDetailToError(
-			errProposalExecutionTimeExpired,
-			ufmt.Sprintf("execute.gno__Execute() || EXECUTION_WINDOW_OVER (now(%d) >= windowEnd(%d))", now, windowEnd),
-		))
-	}
+func makeHandlerKey(pkgPath, function string) string {
+	return ufmt.Sprintf("%s:%s", pkgPath, function)
+}
 
-	yeaUint := proposal.Yea.Uint64()
-	nayUint := proposal.Nay.Uint64()
-	quorumUint := proposal.QuorumAmount
+///////////////////// EXECUTION /////////////////////
+// region: Execute
 
-	if yeaUint < quorumUint {
-		panic(addDetailToError(
-			errProposalQuorumNotSatisfied,
-			ufmt.Sprintf("execute.gno__Execute() || QUORUM_NOT_MET (yes(%d) < quorum(%d))", yeaUint, quorumUint),
-		))
-	}
+type ExecutionContext struct {
+	ProposalId  uint64
+	Now         uint64
+	Config      *Config
+	Proposal    *ProposalInfo
+	WindowStart uint64
+	WindowEnd   uint64
+}
 
-	if yeaUint < nayUint {
-		panic(addDetailToError(
-			errMoreNoVotesThanYesVotes,
-			ufmt.Sprintf("execute.gno__Execute() || NO_MAJORITY (yes(%d) < no(%d))", yeaUint, nayUint),
-		))
+func Execute(proposalId uint64) error {
+	ctx, err := prepareExecution(proposalId)
+	if err != nil {
+		panic(err)
 	}
 
-	if proposal.ProposalType == "COMMUNITY_POOL_SPEND" {
-		// check if the token is registered to community pool
-		registered := cp.GetRegisteredTokens()
-		if !contains(registered, proposal.CommunityPoolSpend.TokenPath) {
-			panic(addDetailToError(
-				errNotRegisteredToCommunityPool,
-				ufmt.Sprintf("execute.gno__Execute() || token(%s) is not registered to community_pool", proposal.CommunityPoolSpend.TokenPath),
-			))
-		}
+	if err := validateVotes(ctx.Proposal); err != nil {
+		panic(err)
+	}
 
-		// trigger community pool spend
-		cp.TransferToken(proposal.CommunityPoolSpend.TokenPath, proposal.CommunityPoolSpend.To, proposal.CommunityPoolSpend.Amount)
+	if err := validateCommunityPoolToken(ctx.Proposal); err != nil {
+		panic(err)
 	}
 
-	if proposal.ProposalType == "PARAMETER_CHANGE" {
-		// trigger parameter change
-		executeParameterChange(proposal.Execution.Msgs)
+	registry := createParameterHandlers()
+	// TODO: must fix this
```

Comment:
handle it.

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/448/commits/33cb763be1e4913decf3205f3418b3fa73466b4c